pkg/network/http/ebpf_gotls.go

```diff
@@ -94,8 +100,25 @@ var structFieldsLookupFunctions = map[bininspect.FieldIdentifier]bininspect.Stru
 }
 
 type GoTLSProgram struct {
-	manager  *errtelemetry.Manager
-	probeIDs []manager.ProbeIdentificationPair
+	manager *errtelemetry.Manager
+
+	// Path to the process/container's procfs
+	procRoot string
+
+	// Process monitor channels
+	procMonitor struct {
+		done   chan struct{}
+		events chan netlink.ProcEvent
+		errors chan error
+	}
+
+	// eBPF map holding the result of binary analysis, indexed by binaries'
+	// inodes.
+	offsetsDataMap *ebpf.Map
+
+	// Internal map holding the result of binary analysis. Used to determine
+	// if analysis is needed when handling a new process' binary.
+	inspected map[uint64]*bininspect.Result
```

Comment:
that's a potential memory leak
assume the scenario where the agent is long living, and we see thousands of short-lived go processes (a.k.a k8s jobs) running for a couple of minutes and then terminating
that map will grow without a limit -> thus we leak memory

2 possible solutions
1.  find the way to know when a process is terminated and clean it
2. use a timed cache (the downside is that we might "rescan" the same binary couple of times)

Reply:
An LRU is also an option, although I think this would be not a big collection given the mapping is from inode -> result.

Reply:
We have settled on a TTL cache, with the following stategy:

- Entries keep track of PIDs associated to a running process, as well as the probes installed on a specific binary.
- When new process spawns, if it is not in the cache, we run the binary analysis then add the entry in the cache and in the eBPF map. No need to keep the result of the analysis afterwards.
- The TTL is infinite as long as there are running processes using that binary.
- Once the last process has stopped, the TTL is set to keep the entries (and the probes) for a few minutes (should this be configurable?).
- If a new process spawns again, we reset the TTL to be infinite. The probes are already there so nothing else needs to be done.
- When the TTL expires we remove the probes, as well as the entry in the offsets_data eBPF map.